### PR TITLE
Add fishing. Add oscillating-meter input. Add fishing-related endings. Misc typing cleanups.

### PR DESCRIPTION
**Pull request by mganderson**
You can load directly into the fishing minigame by entering "fishing" at the `Press enter to continue...` title screen prompt.

In very rare circumstances, it's possible to get one of two fishing-related endings.

A new input method `get_input_from_oscilating_meter()` is added to `engine/printer`

### CODE REVIEW

File: engine/printer/oscilating_input.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import datetime
+import sys
+import threading
+import time
+
+
+class OscillatingMeterPrompt(threading.Thread):
+    """Display a meter that rises and falls every 2 seconds."""
```
Comment (mganderson) [inline engine/printer/oscilating_input.py:8]: In a perfect world, the oscillation speed would be configurable.  Maybe later.

File: world/office/__init__.py [hunk 2]
```diff
@@ -1,15 +1,15 @@
-from engine.world.room_action import RoomAction
 from engine.world.place import Place
 from engine.world.room import Room
-
+from engine.world.room_action import RoomAction
```
Comment (mganderson) [inline world/office/__init__.py:3]: Ugh, the autoformatter went crazy in this file.

File: world/office/__init__.py [hunk 3]
```diff
@@ -18,11 +18,14 @@ def _get_room_actions(actions: list[RoomAction]) -> list[RoomAction]:
     def get_base_actions():
         base_actions = [
             RoomAction(name="Coffee run", scene=CoffeeScene()),
-            RoomAction(name="Declare Whodunit!", scene=DelcareMurdererScene())
+            RoomAction(name="Declare Whodunit!", scene=DelcareMurdererScene()),
+            RoomAction(name="Go fishing", scene=FishingScene()),
```
Comment (mganderson) [inline world/office/__init__.py:22]: This is the only meaningful change in the file.

**Review by AlexTaber — COMMENTED** [review]
I NEED A 🐟 BIG CHUNGUS 🐟  EMOJI

**Review by AlexTaber — APPROVED** [review]

**Pull request merged**